test/cleaners/primitives.test.js

```diff
@@ -26,7 +24,7 @@ describe('asNumber', function () {
   it('accepts only numbers', function () {
     expect(asNumber(0)).equals(0)
     expect(asNumber(-1)).equals(-1)
-    expect(asNumber(NaN)).is.NaN
+    expect(asNumber(NaN)).deep.equals(NaN)
```

Comment:
`deep.equals`?

Reply:
Regular `.equals` uses `===`, which doesn't work with `NaN`. The `deep.equals` uses a much fancier algorithm that does handle `NaN` correctly.

Reply:
Gotta love DSLs. Thanks for the explainer.